javascript/organizing_your_javascript_code/factory_functions_and_module_pattern.md

```diff
@@ -1,305 +1,293 @@
+
+Take the calculator example into consideration. It's very easy to imagine a scenario where you can accidentally create multiple functions with the name `add`. What does `add` do - does it simply add two numbers? Strings? Does it take its input directly from the DOM and display the result? What would you name the functions that do these things? Instead, we can easily encapsulate them inside a module called `calculator` which generates an object with that name, allowing us to explicitly call `calculator.add(a, b)` or `calculator.sub(a, b)`.
+
+### Assignment
 
-- Read up about IIFE's in [this article](http://adripofjavascript.com/blog/drips/an-introduction-to-iffes-immediately-invoked-function-expressions.html). The concept is simple: write a function, wrap it in parentheses, and then immediately call the function by adding `()` to the end of it.
-- An example of creating and using a module pattern: [JavaScript Module Pattern Basics](https://coryrylan.com/blog/javascript-module-pattern-basics).
-- Additional example of creating and using a module pattern: [Module pattern in JavaScript](https://dev.to/tomekbuszewski/module-pattern-in-javascript-56jm).
-- For those who prefer video lessons, here is an excellent YouTube series on modular JS that covers most of the content in this guide: [Modular Javascript](https://www.youtube.com/playlist?list=PLoYCgNOIyGABs-wDaaxChu82q_xQgUb4f).
+<div class="lesson-content__panel" markdown="1">
 
-In our calculator example above, the function inside the IIFE is a simple factory function, but we can just go ahead and assign the object to the variable `calculator` since we aren't going to need to be making lots of calculators, we only need one. Just like the factory example, we can have as many private functions and variables as we want, and they stay neatly organized, tucked away inside of our module, only exposing the functions we actually want to use in our program.
+1. WesBos has a beautiful and in-depth section on scopes and closures. Please check out these sections under "Module 3 - The Tricky Bits":
+  - [The article on scope](https://wesbos.com/javascript/03-the-tricky-bits/scope)
+  - [The article on closures](https://wesbos.com/javascript/03-the-tricky-bits/closures)
+1. [This article](https://tsherif.wordpress.com/2013/08/04/constructors-are-bad-for-javascript/) goes into depth, discussing the problems with constructors as well as suggesting factories as a solution.
+1. [This article](https://dev.to/tomekbuszewski/module-pattern-in-javascript-56jm) provides more details about module pattern with examples.
```

Comment:
Could you tweak the link text on these to be a bit more a11y friendly?

Reply:
Done!